File: 1cpp/Circle.cpp
//
// Created by olile on 02.04.24.
//
#include <stdexcept>

#include "Circle.h"
#include "Square.h"
#include <cmath>


Circle::Circle(double radius) noexcept(false){
    if(radius<0){
        throw std::invalid_argument("Radious must be positive.");
    }
    this->radius = radius;
}

double Circle::calcArea() {
    return M_PI*this->radius*this->radius;
}

double Circle::calcCircumference() {
    return 2*M_PI*this->radius;
}

std::string Circle::info() {
    return "circle";
}

File: 1java/1java/src/Circle.java
// Circle.java (Example implementation)
public class Circle implements Shape {
    private double radius;

    public Circle(double radius) {
        this.radius = radius;
    }

    @Override
    public double calcArea() {
        return Math.PI * radius * radius;
    }

    @Override
    public double calcCircumference() {
        return 2 * Math.PI * radius;
    }

    @Override
    public String info() {
        return "Circle with radius " + radius;
    }
}
